Textile/Textile.sublime-syntax

```diff
@@ -1,225 +1,1110 @@
+  textile-unordered-list-item:
+    - meta_content_scope: markup.list.unnumbered.textile
+    - include: textile-empty-line-pop
+    - match: ^(?=[-*#])
+      pop: 1
+    - include: html
+
+  textile-ordered-list-item:
+    - meta_content_scope: markup.list.numbered.textile
+    - include: textile-empty-line-pop
+    - match: ^(?=[-*#])
+      pop: 1
+    - include: html
+
+  textile-definition-list-item:
+    - meta_content_scope: markup.list.definition.textile
+    - include: eol-pop
+    - include: textile-definitions
+    - match: (.*)\n?
+      captures:
+        1: invalid.illegal.expected-definition.textile
+      pop: 1
+
+  textile-definitions:
+    - match: (?:[[:blank:]]*)([[:upper:]]{3,})(?:[[:blank:]]*)(:=)
+      captures:
+        1: meta.definition.term.textile
+        2: punctuation.separator.key-value.textile
+      push: textile-definition-definition
+
+  textile-definition-definition:
+    - meta_content_scope: meta.definition.definition.textile
+    - match: (=:)(.*)?\n?
+      captures:
+        1: punctuation.definition.definition.end.textile
+        2: invalid.illegal.expected-eol.textile
+      pop: 1
+    - include: textile-empty-line-pop
+    - match: ^(?=-)
+      pop: 1
+    - include: html
+
+###[ INLINE ]##################################################################
+
+  textile-inlines:
+    - include: textile-inline-notextile
+    - include: textile-inline-textile-specials
+    - include: textile-inline-reference
+    - include: textile-inline-link
+    - include: textile-inline-image
+    - include: textile-inline-table-row
+    - include: textile-inline-markup
+    - include: textile-inline-abbreviations
+
+    - match: (\[)(\d+)(\])
+      scope: meta.link.reference.footnote.textile
+      captures:
+        1: punctuation.definition.link.begin.textile
+        2: meta.link.reference.literal.footnote-id.textile
+        3: punctuation.definition.link.end.textile
+
+  textile-inline-notextile:
+    - match: (==)([ ]+)
+      captures:
+        1: punctuation.definition.notextile.begin.textile
+      push: textile-inline-notextile-body
+
+  textile-inline-notextile-body:
+    - meta_scope: meta.notextile.textile
+    # backreference to use whatever is in the match
+    - match: \2(==)
+      captures:
+        1: punctuation.definition.notextile.end.textile
+      pop: 1
+    - include: textile-illegal-newline-pop
+    - include: html
+
+  textile-inline-textile-specials:
+    # https://textile-lang.com/doc/unicode-symbols
+
+    - match: \((?:tm|R|C|1/4|1/2|3/4|o|\+/\-)\)
+      scope: constant.character.textile
+
+    # NOTE: match here to avoid text.html.basic from flagging it
+    - match: \&(?!#[[:alnum:]]+;)
+      scope: markup.raw.inline.ampersand.textile
+
+    - match: (?<=\d[ ])x(?=[ ]\d)
+      scope: markup.raw.inline.dimension-sign.textile
+
+    - match: --
+      scope: markup.raw.inline.long-dash.textile
+
+    - match: (?<=[ ])-(?=[ ])
+      scope: markup.raw.inline.unicode-dash.textile
```

Comment:
As we don't consume or scope numbers or whitespace in general, it might be safe to change those lookbehind containing patterns to something like...

```suggestion
    - match: \d+(?:\.\d+)?[ ]+(x)(?=[ ]+\d)
      captures:
        1: markup.raw.inline.dimension-sign.textile

    - match: '[ ]+(-)(?=[ ])'
      captures:
        1: markup.raw.inline.unicode-dash.textile

    - match: --
      scope: markup.raw.inline.long-dash.textile
```

The first pattern `\d+(?:\.\d+)?` consumes whole floats as single token, but maybe `\d+` would be enough.

Reply:
Maybe we should leave it as is for now.